Sources/Networking/Core/RetryConfiguration.swift

```diff
@@ -8,48 +8,50 @@
 import Foundation
 
 /// Retry of API calls allows various options wrapped into `RetryConfiguration` struct.
-public struct RetryConfiguration {
+public struct RetryConfiguration: Sendable {
     /// The number of retries.
     let retries: Int
     /// The delay between each retry to avoid overwhelming API.
     let delay: DelayConfiguration
     /// A handler which determines wether a request should be retried or not based on an error.
     /// By default errors with status codes `HTTPStatusCode.nonRetriableCodes` are not being retried.
-    let retryHandler: (Error) -> Bool
+    let retryHandler: @Sendable (Error) -> Bool
 
     public init(
         retries: Int,
         delay: DelayConfiguration,
-        retryHandler: @escaping (Error) -> Bool) {
+        retryHandler: @Sendable @escaping (Error) -> Bool) {
         self.retries = retries
         self.delay = delay
         self.retryHandler = retryHandler
     }
     
     // default configuration ignores
-    public static var `default` = RetryConfiguration(
-        retries: 3,
-        delay: .constant(2)
-    ) { error in
-        /// Do not retry authorization errors.
-        if error is AuthorizationError {
-            return false
-        }
-        
-        /// But retry certain HTTP errors.
-        guard let networkError = error as? NetworkError,
-              case let .unacceptableStatusCode(statusCode, _, _) = networkError
-        else {
-            return true
-        }
+    public static var `default`: RetryConfiguration {
+        .init(
+            retries: 3,
+            delay: .constant(2)
+        ) { error in
+            /// Do not retry authorization errors.
```

Comment:
The common practice for comments inside a func scope (eg. inline) is just `//`, not `///`

`///` should be used for documentation of a func, property, etc., not for a single-line comment 😌 

Described for example here: https://microsoft.github.io/swift-guide/Comments.html